---
title: "Figure 7"
author: "Johan"
date: "01/04/2024"
output:
  html_document:
    toc: yes
    toc_depth: 3
    toc_float: yes
    code_folding: hide
    df_print: paged


---






```{r setup, include=FALSE}

repositoryPath = "~/git/Rising-Lab/MajorGland_paper"

#source("~/git/RNAmappingPipeline/R/ExpressionAnalysisFunctions.R")
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(Seurat)
library(knitr)
#install.packages("kableExtra")
#library("viridis")     

library(RColorBrewer)
#library(GenomicFeatures)
library(DESeq2)

library(ropls)

library(ggforce)
library(patchwork)


rename <- dplyr::rename


source("../loadGeneInfo.R")
source("../colorSchemes.R")
source("../customFunctions.R")
summarise = dplyr::summarise               
select = dplyr::select
group_by = dplyr::group_by


```



## Parameters
```{r Parameters, message=FALSE, warning=FALSE}

minExpressionCutoff =20
DataGenerated = TRUE

orderSilkGroup = c("MaSp1","MaSp2","MaSp3","MaSp4","AmSp-like","Spice-LMa","Not in silk")
```



## Gene information
```{r Gene information, message=FALSE, warning=FALSE}

geneInfo =  getGeneInfoFull(repositoryPath = repositoryPath) %>%
  select(gene_id, geneName, lascID,scGeneName,Chr,Start, End)
GI = getGeneInfoFull(repositoryPath = repositoryPath)
geneInfo.transcript = addTranscriptInfo(geneInfo = GI,repositoryPath = repositoryPath) %>%group_by(gene_id) %>% 
  arrange(-tx_len ) %>% slice_head(n = 1) %>% select(gene_id,geneName,tx_name,tx_len)

proteinInfo = getMajorAmpullateGlandProteinsSC(repositoryPath = repositoryPath)  %>%
  mutate(silkGroup = factor(silkGroup,levels =orderSilkGroup))

geneInfo = getGeneInfoFull(repositoryPath)
transcriptInfo = addTranscriptInfo(repositoryPath = repositoryPath,geneInfo = geneInfo) %>%
  group_by(gene_id) %>%
  arrange(-tx_len) %>%
  slice_head(n = 1) %>% ungroup()



merged_df <- read.csv("../../annotation/proteomics/proteinComposition.csv") %>%
  mutate(tx_name = Protein.ID) %>%
  inner_join(transcriptInfo) %>%
  select(Length,geneName) %>% inner_join(proteinInfo) ->
  proteinInfo
```



```{r Layers analysis, message=FALSE, warning=FALSE}


UreaSilkFile =  "../../data/proteomics/majorSilk/Samples Report With Clusters for Samples_7_15.xls"

old = read_tsv(UreaSilkFile,skip = 49) %>%
  rename(tx_name = "Accession Number") %>%
  inner_join(geneInfo.transcript) %>%
  select(geneName,10:18) 

new = read_tsv(UreaSilkFile,skip = 49) %>%
  rename(geneName = "Accession Number") %>%
  inner_join(geneInfo.transcript) %>%
  select(geneName,10:18) 

UreaSilk = bind_rows(old,new) %>%
  pivot_longer(-geneName, names_to = "sample", values_to = "Percentage_of_Total_Spectra") %>%
  mutate(Percentage_of_Total_Spectra = gsub("%","",Percentage_of_Total_Spectra))%>%
  mutate(Percentage_of_Total_Spectra = as.numeric(gsub(",",".",Percentage_of_Total_Spectra)))
  
UreaSilk %>% 
  dplyr::group_by(sample ) %>%
  dplyr::summarise(TotalPercentage_of_Total_Spectra = sum(Percentage_of_Total_Spectra)) %>%
  inner_join(UreaSilk) %>%
  mutate(fractionMatches = Percentage_of_Total_Spectra/TotalPercentage_of_Total_Spectra) ->
UreaSilk

UreaSilk.metaData = data.frame (sample = c(paste("Sample",7:15, sep = " "))) %>%
  mutate(type = c(rep("silk",9))) %>%
  mutate(solution = c(rep("UREA_2M",3),rep("UREA_4M",3),rep("UREA_8M",3))) %>%
  mutate(Urea = c(rep("2M",3),rep("4M",3),rep("8M",3))) %>%
  mutate(rep = c(rep(1:3,3)))
  

```
```{r Layers analysis 2, message=FALSE, warning=FALSE}


protein.urea = UreaSilk %>% inner_join(UreaSilk.metaData) %>% inner_join(proteinInfo)

protein.urea %>%
  group_by(sample) %>%
  summarise(totalFraction = sum(fractionMatches)) %>%
  inner_join(protein.urea) %>%
  mutate(percent = fractionMatches/totalFraction*100)%>%
  inner_join(proteinInfo)->
  protein.urea

protein.urea %>% group_by(geneNameProtein,solution) %>%
  summarise(mean = mean(percent))




protein.urea %>%
  group_by(geneNameProtein) %>% 
  summarise(percent_mean = mean(percent), percent_sd = sd(percent)) %>%
  inner_join(protein.urea) %>%
  mutate(zscore = (percent- percent_mean)/percent_sd)->
protein.urea
```
# Figure 7E urea concentrations  
```{r Figure 7 urea concentrations, message=FALSE, warning=FALSE}
protein.urea %>% 
  group_by(geneNameProtein, solution) %>%
  summarise(zscore = mean(zscore)) ->
  protein.urea.zscore

protein.urea.zscore %>%
  mutate(geneNameProtein = factor(geneNameProtein, levels =geneOrder3 )) %>%
  mutate(solution = factor(solution, levels =c("UREA_8M","UREA_4M","UREA_2M"))) %>%
  ggplot(aes(geneNameProtein, y = solution,, fill =zscore ))+  
  geom_tile(color = "white",
            lwd = 1.5,
            linetype = 1) +
  scale_fill_gradientn(colours = c("blue","white","red")) + 
  theme_light() +  
 theme(axis.text.x=element_text(angle = -90, hjust = 0)) -> 
  urea.distributions
  

urea.distributions
ggsave("../../figures/figure7/figure7E.pdf", width = 8, height = 3,plot = urea.distributions)



```



## Figure 7D scRNA

```{r}


major.sct.integrated = readRDS("../../data/scRNA/majorGland/scRNA.major.integrated.2024.rds")

major.sct.integrated = SetIdent(major.sct.integrated, value ="clusterZones")
AverageExpression(major.sct.integrated)   ->
  scE 

scE$SCT %>%
  as.data.frame() %>%
  rownames_to_column(var = "scGeneName") %>%
  pivot_longer(-scGeneName,
               names_to = "cluster",
               values_to = "expression") %>%
  inner_join(proteinInfo) %>%
  select(geneNameProtein,cluster,expression)   %>%
  mutate(cluster = factor(cluster, levels = rev(clusterOrder))) ->
  sc.sct.expression.orig


Markergenes.sc.all.sctransform  = read_csv("../../annotation/scRNA/Major/Markers.SCT.integrated.majorCluster.18PIS.2024.csv")

Markergenes.sc.all.sctransform %>% 
  mutate(MarkerGene = "Yes") %>%
  full_join(sc.sct.expression.orig) %>%
  mutate(MarkerGene = replace(MarkerGene,is.na(MarkerGene),"No")) ->
sc.sct.expression  

  

sc.sct.expression %>%
  mutate(lg2Exp = log2(expression+1)) %>%
  group_by(geneNameProtein) %>%
  summarise(mean = mean(expression), std = sd(expression)) %>%
  inner_join(sc.sct.expression) %>%
  mutate(lg2Exp = log2(expression+1)) %>%
  mutate(zscore = (expression-mean)/std) %>%
  mutate(geneNameProtein = factor(geneNameProtein, levels = geneOrder3))  %>%
  mutate(cluster = factor(cluster, levels =clusterOrder ))->
    sc.sct.expression
  #%>% mutate(MarkerGene = factor(MarkerGene, levels = c("Yes","No"))) 


MarkerColors = c("Yes" = "#111111ff", "No" = "#ffffff00") 

sc.sct.expression %>% 
  filter(cluster %in% c("ZoneA_MaSp2" ,"ZoneA_MaSp1", "ZoneA_SpiCE-LMa3",
     "ZoneB_MaSp3","ZoneC_SpiCE-LMa1","ZoneABC") ) -> sc.sct.expression

ggplot(data = sc.sct.expression,
       mapping = aes(x = geneNameProtein, 
                     y = cluster, 
                     fill = zscore,
                     color = MarkerGene )) + 
  geom_tile(lwd = 0.5,
            linetype = 1,width=0.8,height = 0.8)+
  theme_light() +
  theme(axis.text.x=element_text(angle = -90, hjust = 0,vjust = 0.5)) +
  scale_fill_gradientn(colors = c("blue", "white", "red"),values = c(0,0.4,1))  +
  scale_colour_manual(values =MarkerColors) ->
  scExpressionGraf


ggsave("../../figures/figure7/figure7D.pdf", width = 8,height = 4,plot = scExpressionGraf)



```





# Figure 7B Image

```{r }
crossSection = read_csv("~/git/Rising-Lab/MajorGland_paper/annotation/spatialData/QUpathCrossSection.csv")




crossSection %>% column_to_rownames("ID2") %>%
  select(Hematoxylin_Mean,Eosin_Mean)-> test

m <- c( "average", "single", "complete", "ward")
names(m) <- c( "average", "single", "complete", "ward")
library(cluster)
library(factoextra)
df = test
ac <- function(x) {
  agnes(df, method = x)$ac
}
sapply(m, ac)
clust <- agnes(df, method = "ward")

pdf("~/Box/bridgeSpider/paper1/main/figures/figure6/eosin_hematoxilin.distribution.tree.pdf")
tree = pltree(clust, cex = 0.6, hang = -1, main = "Dendrogram") 
dev.off()
nrOfClusters = gap_stat <- clusGap(df, FUN = hcut, nstart = 25, K.max = 8, B = 50 )
#produce plot of clusters vs. gap statistic
pdf("~/Box/bridgeSpider/paper1/main/figures/figure6/eosin_hematoxilin.distribution.tnrOfClusters.pdf")
fviz_gap_stat(gap_stat)
dev.off()

d <- dist(df, method = "euclidean")



#perform hierarchical clustering using Ward's method
final_clust <- hclust(d, method = "ward.D2" )
#cut the dendrogram into 4 clusters


geneName = names(cutree(final_clust, 3))

crossSection$class = cutree(final_clust, 3)
clusters = cutree(final_clust, 3)

library(ComplexHeatmap)
  
row_ha_left = rowAnnotation(Class = crossSection$Class,
                            Region = crossSection$Region,
                            Image = crossSection$Image,
                            col = list(Class = AllColors2,
                                       Region = AllColors2,
                                       Image = AllColors2))


as.data.frame(clusters) %>% 
  rownames_to_column(var = "ID2")->
  clusters

crossSection = inner_join(crossSection,clusters)

pdf("../../supplementaryFigures/SuppFig40/SuppFig40.hierarchicalCluster.pdf", width = 2.6, height = 6)
ComplexHeatmap::Heatmap(crossSection[,c("Hematoxylin_Mean","Eosin_Mean")],
                        row_split = crossSection$class, cluster_columns = FALSE,
                        cluster_rows = TRUE, 
                         left_annotation = row_ha_left)

dev.off()


ggplot(crossSection )+
  coord_cartesian(ylim =c(0, 0.5),xlim =c(0, 0.4)) + 
  geom_point(aes(x =Hematoxylin_Mean, y = Eosin_Mean,
                        color = Class,
                        shape = Region ),size = 2)+
  theme_light()+
  scale_shape_manual(values=c(0,19)) +
  scale_color_manual(values = AllColors2)  ->
  class

 ggplot(crossSection)+
  coord_cartesian(ylim =c(0, 0.5),xlim =c(0, 0.4)) + 
  geom_point(aes(x =Hematoxylin_Mean, y = Eosin_Mean,
                        color = as.factor(clusters)))+
  theme_light()+
  scale_color_manual(values = AllColors2) ->
  cluster
 
 class+ cluster + plot_layout(nrow = 2)
 
 ggsave("../../supplementaryFigures/SuppFig40/SuppFig40.hierarchicalCluster.pdf", width = 5, height = 3)

 
 
 ggplot(crossSection )+
  coord_cartesian(ylim =c(0, 0.5),xlim =c(0, 0.4)) + 
  geom_point(aes(x =Hematoxylin_Mean, y = Eosin_Mean,
                        color = Class),size = 2)+
  theme_light()+
  scale_shape_manual(values=c(0,19)) +
  scale_color_manual(values = AllColors2) + facet_wrap(facets = "Region") ->
   fig7b
 ggsave("../../figures/figure7/figure7B.pdf", width = 8, height = 4, plot  = fig7b)

  

 
```
